core/spec.py

```
@@ -0,0 +1,72 @@
+        return {
+            "task": self.task,
+            "created_at": self.created_at,
+            "status": self.status,
+            "steps_executed": self.steps_executed,
+            "tools_used": self.tools_used,
+            "artifacts": self.artifacts,
+            "decisions": self.decisions,
+            "errors": self.errors,
+        }
+
+    def save(self, filepath: str = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)